### ISSUE TITLE
UWP exported app does not work in release mode

### ISSUE THREAD

**Issue by vnen** [godotengine/godot#9131]
I'm having a hard time figuring out this problem, so I'll leave this issue open until I (or someone else) can solve it. This is valid for 2.1 and 2.2, so I assume it is the same for master.

The UWP app exported does not run when installed, it just crashes on opening. The only debug info I have is the EventViewer, which does not help at all:

`ActivateApplicationForContractByAppIdAsUserWithHost of the app Godot.Engine.Test_q1xfrwf3pzxew!App for the Windows.Launch contract failed with Access is denied..`

Trying to debug the installed app from Visual Studio (via Debug > Other Debug Targets > Debug Installed App Package) is not helpful because it runs fine there! So there's something fishy about how apps are launched that is a mystery (at least for me).

Summing up what I know:

1. The app runs fine if compiled with `target=debug` or `target=release_debug` build, but crashes on opening with `target=release`.
2. The `target=release` build runs fine from Visual Studio, even when debugging the installed app.
3. Compiling the release with `/SUBSYSTEM:CONSOLE` instead of `/SUBSYSTEM:WINDOWS` makes it work, but also opens the console window when launched.
4. This is not the case when running on the phone emulator. The release build opens fine there.

**Comment by nunodonato** [comment]
Have you tried running Process Monitor to understand what might be happening?

**Comment by nunodonato** [comment]
Just to add a few more info from my experience: my game *always* crashes at a particular point in the UWP build (no crash in other platforms), both with release and release_debug, but NOT with target=debug

**Comment by vnen** [comment]
@nunodonato that crash is something in the audio server. I am not sure if it's related to the server itself or my xaudio driver implementation. And I don't know why it does not crash in debug builds.

**Comment by nunodonato** [comment]
@vnen I've been away for a while. Was there any significant changes done recently that could have improved this?

**Comment by vnen** [comment]
@nunodonato no, I don't think so. I'll have time to investigate next week.

**Comment by Corruptinator** [comment]
This is from the recent Godot 3.0 Beta 2:

Not sure if this is any help but I've was able to figure out to export a game project to UWP, so I've tried to export a 3D model with a texture to UWP. However, as you mentioned in this issue, it crashes on Release mode. So I tried Debug and it was able to open fine but with a bit of a big issue:

<img width="769" alt="texture_error_uwp" src="https://user-images.githubusercontent.com/5027750/34318235-eeefaf6a-e777-11e7-9a50-ffece486112f.png">

The texture does not load on the sphere.(Its the Godot Icon PNG file I used to test)

The Debug output terminal also generated this Error message before the textureless sphere was displayed:

`**ERROR**: Condition ' !is_inside_tree() ' is true. returned: Transform()
   At: scene\3d\spatial.cpp:265:Spatial::get_global_transform() - Condition ' !is_inside_tree() ' is true. returned: Transform()`

Just figure I'd share what problems I've ran into, once again not sure if this helps at all.

**Comment by six519** [comment]
Any news on this bug? Is it already fixed on version 3?

**Comment by StormtekGames** [comment]
Is there any workaround for this bug? This still does not work in 3.1-beta1...

**Comment by reduz** [comment]
My feeling is that this won't work best until we port to Vulkan and then
use gfx-rs or similar to layer over DirectX12

**Comment by vnen** [comment]
@StormtekGames 
> Is there any workaround for this bug? This still does not work in 3.1-beta1...

The only workaround I know is on the original post:

> Compiling the release with `/SUBSYSTEM:CONSOLE` instead of `/SUBSYSTEM:WINDOWS` makes it work, but also opens the console window when launched.
